docs/source/tools/export.rst

```diff
@@ -0,0 +1,164 @@
+
+The flags ``--host`` and ``--port`` define the Daml ledger to connect to. You
+can omit these flags if you are invoking the command from within a Daml project
+with a running ledger, e.g. with a running ``daml start``.
+
+The ``--party`` flags define which contracts will be included in the export. In
+the above example only contracts visible to the parties Alice and Bob will be
+included in the export. Lack of visibility of certain events may cause exports to fail, see :ref:`caveats <export-caveats>`.
+
+The ``--output`` flag defines the directory prefix under which to generate the
+Daml project that contains the Daml script that represents the ledger export.
+The flag ``--sdk-version`` defines which Daml Connect version to configure in
+the generated ``daml.yaml`` configuration file.
+
+By default an export will reproduce all transactions in the ledger history. The
+:ref:`ledger offsets <export-ledger-offsets>` section describes how to change
+this behavior.
+
+Output
+******
+
+The generated Daml code contains the following top-level definitions:
+
+``data Parties``
+  A record that holds the relevant Daml parties.
+``allocateParties : Script Parties``
+  A Daml script that allocates fresh parties on the ledger and returns them in
+  the ``Parties`` record.
+``export : Parties -> Script ()``
+  The Daml ledger export encoded as a Daml script. Given the relevant parties
+  this script will reproduce the ledger state when executed. You can read this
+  script to understand the exported ledger state or history, and you can modify
+  this script for debugging or testing purposes.
+``testExport : Script ()``
+  A Daml script that will first invoke ``allocateParties`` and then ``export``.
+
+In most simple cases the generated Daml script will use the functions
+``submit`` or ``submitMulti`` to issue ledger commands that reproduce a
+transaction or ACS. In some cases the generated Daml script
+will fall back to the more general functions ``submitTree`` or
+``submitTreeMulti``.
+
+For example, the following generated code issues a create-and-exercise command
+that creates an instance of ``ContractA`` and exercises the choice ``ChoiceA``.
+The function ``submitTree`` returns a ``TransactionTree`` object that captures
+all contracts that are created in the transaction. The ``fromTree`` function is
+then used to extract the contract ids of the ``ContractB`` contracts that were
+created by ``ChoiceA``.
+
+.. code-block:: daml
+
+  tree <- submitTree alice_0 do
+    createAndExerciseCmd
+      Main.ContractA with
+        owner = alice_0
+      Main.ChoiceA
+  let contractB_1_1 = fromTree tree $
+        exercised @Main.ContractA "ChoiceA" $
+        created @Main.ContractB
+  let contractB_1_2 = fromTree tree $
+        exercised @Main.ContractA "ChoiceA" $
+        createdN @Main.ContractB 1
+
+.. TODO[AH] Add a full example project and example export.
```

Comment:
This could be expanded to a more complete example that includes an initial Daml script to create an example history and then display the resulting Daml script exports for different configurations. This could be imported via `literalinclude` from a working example similar to how other documentation chapters do, to make sure that the documentation stays up-to-date and correct. However, this requires a bit more work, so I'll leave that for a separate PR.